### DISCUSSION TITLE
Sparse Fieldsets

### DISCUSSION THREAD

**Discussion by olragon** [tailflow/laravel-orion#296] [Ideas]
https://jsonapi.org/format/#fetching-sparse-fieldsets
> A client MAY request that an endpoint return only specific fields in the response on a per-type basis by including a fields[TYPE] parameter.

Can we implement this usage pattern?

**Comment by alexzarbn** [comment]
Hi @olragon,

It is currently not supported and there is no scheduled milestone for this feature yet. Leaving the issue open to see, if more developers would need that functionality.

**Comment by glennjacobs** [comment]
This would be really handy to have. Would you be open to a PR?

**Comment by alexzarbn** [comment]
Hi @glennjacobs,

Sure, PRs are always welcome ~

**Comment by rossity** [comment]
@alexzarbn 

Just want to also add that this feature would be immensely useful, especially if you add support in the JS library. Would cut down a lot on having to write different controllers/resources/or other logic that removes certain fields from a search.

**Comment by emeGuan** [comment]
Maybe someone might find it interesting.
https://github.com/emeGuan/laravel-resources
It is not a substitute for this fantastic Laravel Orion library, far from it. But it can be useful for someone who wants to build their API using https://laravel.com/docs/10.x/eloquent-resources
Regards.

**Comment by rossity** [comment]
@alexzarbn @olragon 

I created a simple implementation of this. 

https://github.com/tailflow/laravel-orion/pull/226
https://github.com/tailflow/laravel-orion-ts/pull/36

Edit: updated them to use sparse fieldsets.

**Comment by cirrusmostar** [comment]
This should be mentioned in the RADME along with other essential functionality that is not available in this package.
